module-utils/utility/Version.cpp

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2017-2024, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "Version.hpp"
+
+#include <module-utils/i18n/JSONLoader.hpp>
+#include <log/log.hpp>
+
+namespace utils
+{
+
+    std::optional<std::string> getPreviousVersionFromFile(const std::filesystem::path &path)
+    {
+        const auto &load = utils::JSONLoader(path);
+        if (not load.has_value()) {
+            LOG_ERROR("Cannot parse %s to json", path.c_str());
+            return {};
+        }
+        const auto &json = load.value();
+        return json["preOsVersion"].string_value();
```

Comment:
I'd name this key `prevOsVersion`, as with current form it sounds to me more like as "version that was there before OS" rather than "version of previous OS" :wink:

Reply:
I'm fine with your proposition ;) Done